pennylane/templates/subroutines/all_singles_doubles.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the AllSinglesDoubles template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+import numpy as np
+import pennylane as qml
+from pennylane.operation import Operation, AnyWires
+from pennylane.ops import BasisState
+
+
+class AllSinglesDoubles(Operation):
+    r"""Apply the particle-conserving :class:`~.pennylane.SingleExcitation` and
```

Comment:
This is not clear enough in saying that the template builds a circuit consisting of ALL such excitation gates, which conserve spin